### PR TITLE
(Feature) #593 apply bonus branch

### PR DESCRIPTION
**Pull request by yaroslav-fedyshyn-nordwhale**
# Description

This PR is base on beta branch. And the base branch is beta too.

Implement request to redeem bonuses from web3 site for whitelisted users only.
Save w3Token inside user object.
Get available bonuses for user. Check if user is white listed. Create transaction for blockchain to award user with bonuses.

Fixes https://app.zenhub.com/workspaces/gooddollar-5d50833888a83c6880d3e345/issues/gooddollar/gooddapp/593

# How Has This Been Tested?

Should be tested on localhost:3000.
The same issue with isVerified function is present ob backend. You should do the same as on Dapp - return true just for test purposes.
There also unit test added for new endpoint.

# Checklist:
- [x] PR title matches follow: (Feature|Bug|Chore) Task Name
- [x] My code follows the style guidelines of this project
- [x] I have followed all the instructions described in the initial task (check Definitions of Done)
- [x] I have performed a self-review of my own code
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] I have added reference to a related issue in the repository
- [x] I have added a detailed description of the changes proposed in the pull request. I am as descriptive as possible, assisting reviewers as much as possible.
- [ ] I have added screenshots related to my pull request ( for frontend tasks)
- [ ] I have pasted a gif showing the feature.
- [x] @mentions of the person or team responsible for reviewing proposed changes

This PR is base on beta branch. And the base branch is beta too.
Link to Dapp PR: https://github.com/GoodDollar/GoodDAPP/pull/728

### CODE REVIEW

File: src/server/verification/verificationAPI.js [hunk 1]
```diff
@@ -395,35 +388,140 @@ const setup = (app: Router, verifier: VerificationAPI, storage: StorageAPI) => {
...
+      }
+
+      if (!wallet_token) {
+        return res.status(400).json({
+          ok: -1,
+          message: 'Missed W3 token'
+        })
+      }
+
+      const isQueueLocked = await txManager.isLocked(currentUser.gdAddress)
```
Comment (sirpy) [inline src/server/verification/verificationAPI.js]: why is this needed?
txManager.lock should be enough no?

Comment (yaroslav-fedyshyn-nordwhale) [reply src/server/verification/verificationAPI.js]: @sirpy I added this for the case when tokens are minting long time in the blockchain. So if Dapp sends the second request for bonuses it just sends a response with the message that the G$ are in mint process. It is made to prevent Dapp requests fails in 30 seconds if there will be no response from server.

**Comment by yaroslav-fedyshyn-nordwhale** [comment]
> you did changes to tx-manager, they dont seem like they fix anything, but removing new functionality

@sirpy yes, you are right. I fixed it in new commit

File: src/server/verification/verificationAPI.js [hunk 2]
```diff
@@ -395,35 +388,140 @@ const setup = (app: Router, verifier: VerificationAPI, storage: StorageAPI) => {
...
+        const w3Data = await W3Helper.getLoginOrWalletToken(currentUser)
+
+        if (w3Data && w3Data.wallet_token) {
+          wallet_token = w3Data.wallet_token
+
+          storage.updateUser({ identifier: currentUser.loggedInAs, w3Token: w3Data.wallet_token })
+        }
+      }
+
+      if (!wallet_token) {
```
Comment (StanislavShevchenko) [inline src/server/verification/verificationAPI.js]: duplicate condition, look 435 line

Comment (yaroslav-fedyshyn-nordwhale) [reply src/server/verification/verificationAPI.js]: @StanislavShevchenko it is not a mistake. First time (in 435 line) I am checking if there is a w3Token in user record. If not, then trying to get it from W3 Site. If there's still not w3Token then the server responded with error to the Dapp.
